gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/purger/HivePurgerExtractor.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance.purger;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+import gobblin.compliance.ComplianceConfigurationKeys;
+import gobblin.compliance.DatasetDescriptor;
+import gobblin.compliance.HivePartitionDataset;
+import gobblin.compliance.HivePartitionFinder;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.Extractor;
+import gobblin.util.reflection.GobblinConstructorUtils;
+
+
+/**
+ * This extractor doesn't extract anything, but is used to instantiate and pass {@link PurgeableHivePartitionDataset}
+ * to the converter.
+ *
+ * @author adsharma
+ */
+public class HivePurgerExtractor implements Extractor<PurgeableHivePartitionDatasetSchema, PurgeableHivePartitionDataset> {
+  private PurgeableHivePartitionDataset record;
+  private boolean read = false;
+
+  public HivePurgerExtractor(WorkUnitState state)
+      throws IOException {
+    this.record = createPurgeableHivePartitionDataset(state);
+  }
+
+  @Override
+  public PurgeableHivePartitionDatasetSchema getSchema() {
+    return new PurgeableHivePartitionDatasetSchema();
+  }
+
+  /**
+   * There is only one record {@link PurgeableHivePartitionDataset} to be read per partition, and must return null
+   * after that to indicate end of reading.
+   */
+  @Override
+  public PurgeableHivePartitionDataset readRecord(PurgeableHivePartitionDataset record)
```

Comment:
Would it make more sense if you move `this.record = createPurgeableHivePartitionDataset(state)` when calling this method for the first time, instead of doing it in the constructor? Since if something wrong happens and `readRecord` never gets called, it is wasteful to call `createPurgeableHivePartitionDataset` in the constructor.